mlrun/api/api/endpoints/workflows.py

```diff
@@ -0,0 +1,423 @@
+                artifact_path=artifact_path,
+            )
+
+            state = mlrun.run.RunStatuses.running
+
+            return {
+                "project": project.metadata.name,
+                "name": workflow_spec.name,
+                "status": state,
+                "run_id": run.uid(),
+            }
+
+    except Exception as err:
+        logger.error(traceback.format_exc())
+        log_and_raise(HTTPStatus.BAD_REQUEST.value, reason=f"runtime error: {err}")
+
+
+@router.get(
+    "/projects/{project}/{uid}", response_model=mlrun.api.schemas.GetWorkflowResponse
+)
+def get_workflow_id(
+    project: str,
+    uid: str,
+    auth_info: mlrun.api.schemas.AuthInfo = fastapi.Depends(
+        mlrun.api.api.deps.authenticate_request
+    ),
+    db_session: Session = fastapi.Depends(mlrun.api.api.deps.get_db_session),
+):
+    """
+    Retrieve workflow id by the uid of the runner.
+    Supporting workflows that executed by the remote engine **only**.
+
+    :param project:     name of the project
+    :param uid:         the id of the running job that runs the workflow
+    :param auth_info:   auth info of the request
+    :param db_session:  session that manages the current dialog with the database
+
+    :return: The id and status of the workflow.
+    """
+    # Check permission READ run:
+    mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.api.schemas.AuthorizationResourceTypes.run,
+        project,
+        uid,
+        mlrun.api.schemas.AuthorizationAction.read,
+        auth_info,
+    )
+    # Reading run:
+    run_db = get_run_db_instance(db_session)
+    run = run_db.read_run(uid=uid, project=project)
+
+    run_object = mlrun.RunObject.from_dict(run)
+
+    status = ""
+    # Does not have results yet:
+    if not isinstance(run_object.status.results, dict):
+        return {"workflow_id": None, "status": status}
+
+    workflow_id = run_object.status.results.get("workflow_id", None)
+
+    # Check permission READ run:
+    mlrun.api.utils.auth.verifier.AuthVerifier().query_project_resource_permissions(
+        mlrun.api.schemas.AuthorizationResourceTypes.run,
+        project,
+        workflow_id,
+        mlrun.api.schemas.AuthorizationAction.read,
+        auth_info,
+    )
+    engine = run_object.status.results.get("engine")
+    if engine:
+        if engine == "kfp":
+            # special case for kfp because _KFPRemoteRunner.get_state() uses get pipeline
+            # which require access to remote api-service
+            pipeline = mlrun.api.crud.Pipelines().get_pipeline(
+                db_session=db_session, run_id=workflow_id, project=project
+            )
+            status = pipeline["run"].get("status", "")
+        else:
+            engine = mlrun.projects.pipelines.get_workflow_engine(engine)
+            status = engine.get_state(workflow_id, project)
+
+    return {"workflow_id": workflow_id, "status": status}
```

Comment:
so actually I am not sure we are aligned with what we talked about how to implement this endpoint, I don't think it needs to also query the pipeline state, I think that for that we should have different endpoint.